cmd/minikube/cmd/delete.go

```diff
@@ -601,49 +602,99 @@ func killMountProcess() error {
 	return nil
 }
 
+// killProcess takes a path to look for a pidfile (space-separated),
+// it reads the file and converts it to a bunch of pid ints,
+// then it tries to kill each one of them.
+// If no errors were ecnountered, it cleans the pidfile
 func killProcess(path string) error {
 	pidPath := filepath.Join(path, constants.MountProcessFileName)
 	if _, err := os.Stat(pidPath); os.IsNotExist(err) {
 		return nil
 	}
-
 	klog.Infof("Found %s ...", pidPath)
-	out, err := os.ReadFile(pidPath)
+
+	ppp, err := GetPids(pidPath)
 	if err != nil {
-		return errors.Wrap(err, "ReadFile")
+		return err
 	}
-	klog.Infof("pidfile contents: %s", out)
-	pid, err := strconv.Atoi(string(out))
-	if err != nil {
-		return errors.Wrap(err, "error parsing pid")
+
+	// we're trying to kill each process, without stopping at first error encountered
+	// error handling is done below
+	var errs []error
+	for _, pp := range ppp {
+		err := trySignalProcess(pp)
+		if err != nil {
+			errs = append(errs, err)
+		}
+
+	}
+
+	if len(errs) == 1 {
+		// if we've encountered only one error, we're returning it:
+		return errs[0]
+	} else if len(errs) != 0 {
+		// if multiple errors were encountered, combine them into a single error
+		out.Styled(style.Failure, "Multiple errors encountered:")
+		for _, e := range errs {
+			out.Err("%v\n", e)
+		}
+		return errors.New("multiple errors encountered while closing mount processes")
 	}
-	// os.FindProcess does not check if pid is running :(
+
+	// if no errors were encoutered, it's safe to delete pidFile
+	if err := os.Remove(pidPath); err != nil {
+		return errors.Wrap(err, "While closing mount-pids file")
+	}
+
+	return err
+}
+
+// trySignalProcess takes a PID as argument and tries to SIGKILL it
+func trySignalProcess(pid int) error {
 	entry, err := ps.FindProcess(pid)
 	if err != nil {
-		return errors.Wrap(err, "ps.FindProcess")
+		return errors.Wrap(err, fmt.Sprintf("ps.FindProcess for %d", pid))
 	}
 	if entry == nil {
 		klog.Infof("Stale pid: %d", pid)
-		if err := os.Remove(pidPath); err != nil {
-			return errors.Wrap(err, "Removing stale pid")
-		}
-		return nil
+		return errors.Wrap(err, fmt.Sprintf("removing stale pid: %d", pid))
 	}
 
 	// We found a process, but it still may not be ours.
```

Comment:
here can we ensure the PID's binary contains the name minikube ?
in normal "ps aux" in terminal we can see the PID have minikube binary there is that campability, I bet go-ps lets us do that too.

we can move that logic to a func called

"doesPIDBelongToMinikube"
and it should ensure that PID 's binary path has word minikube in it